Singletons/OnDemandHolderSingleton.java

```diff
@@ -0,0 +1,13 @@
+package Singletons;
+
+//this is the best solution
+//class creation in java is thread-safety, that means when class creates with an object, it is 100% thread-safe
+public class OnDemandHolderSingleton {
+    public static class SingletonHolder {
```

Comment:
I think, better to use private to limit access options.
```suggestion
    private static class SingletonHolder {
```